builder/common/step_run_source_instance.go

```diff
@@ -229,6 +232,24 @@ func (s *StepRunSourceInstance) Run(ctx context.Context, state multistep.StateBa
 		}
 	}
 
+	if s.CapacityReservationPreference != "" {
```

Comment:
If it's not empty, should we check that the value is either `none` or `open`? Or will a request fail when we set something invalid?

Reply:
Also, I imagine it being unset is the same as setting it to `none`? If not we should probably replace the empty value with `none` and set it in the configuration

Reply:
Ah good catch! Thanks for the review.